### PR TITLE
Fix potential NPE in UsersTool

### PR DESCRIPTION
**Pull request by cbuescher**
I found the first UserException thrown in #execute marked as dead code when
accidentally switching this from a warning to an error. It looks like the output
of FileUserPasswdStore.parseFile a line above shouldn't be wrapped into another
map since its output can be null. Looking at Commit fab5e21e7d I wonder what
changed this back? The history of this file is a bit unclear to me, but I think
L225 should be changed.

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 2

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/file/tool/UsersTool.java [hunk 1]
```diff
@@ -221,7 +222,7 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
 
             Path file = FileUserPasswdStore.resolveFile(env);
             FileAttributesChecker attributesChecker = new FileAttributesChecker(file);
-            Map<String, char[]> users = new HashMap<>(FileUserPasswdStore.parseFile(file, null, env.settings()));
+            Map<String, char[]> users = FileUserPasswdStore.parseFile(file, null, env.settings());
             if (users == null) {
                 throw new UserException(ExitCodes.CONFIG, "Configuration file [" + file + "] is missing");
```
Comment (cbuescher) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/file/tool/UsersTool.java]: Currently this looks dead. Furthermore, FileUserPasswdStore.parseFile can return null which I think will then create an NPE in the HashMap ctor.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/file/tool/UsersTool.java [hunk 2]
```diff
@@ -221,14 +222,15 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
...
             }
             if (users.containsKey(username) == false) {
                 throw new UserException(ExitCodes.NO_USER, "User [" + username + "] doesn't exist");
             }
             final Hasher hasher = Hasher.resolve(XPackSettings.PASSWORD_HASHING_ALGORITHM.get(env.settings()));
+            users = new HashMap<>(users); // make modifiable
```
Comment (cbuescher) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/file/tool/UsersTool.java]: same approach as in L129

**Comment by cbuescher** [comment]
@elasticmachine run the gradle build tests 2

**Comment by cbuescher** [comment]
@elasticmachine run elasticsearch-ci/2

**Review by jaymode — APPROVED** [review]
LGTM. Nice catch

**Comment by jaymode** [comment]
@cbuescher this also looks like a issue in 6.x. Do you mind backporting to 6.x and 6.6?

**Pull request merged**